src/Utilities/hooks/useFetchOperatingSystems.js

```diff
@@ -1,32 +1,45 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 import { getOperatingSystems } from '../../api';
 
+const initialState = {
+  operatingSystems: [],
+  operatingSystemsLoaded: false,
+};
+
 const useFetchOperatingSystems = ({
   apiParams = [],
   hasAccess,
   showCentosVersions,
+  fetchCustomOSes,
 }) => {
   const mounted = useRef(true);
-  const initialState = {
-    operatingSystems: [],
-    operatingSystemsLoaded: false,
-  };
+
   const [data, setData] = useState(initialState);
 
   const fetchOperatingSystems = useCallback(async () => {
-    if (typeof hasAccess === 'undefined') {
-      return await getOperatingSystems(apiParams, showCentosVersions);
-    } else if (typeof hasAccess !== 'undefined' && hasAccess === true) {
-      return await getOperatingSystems(apiParams, showCentosVersions);
+    if (hasAccess === false) return;
+
+    const fetchFn = fetchCustomOSes || getOperatingSystems;
+    const fetchArgs = [apiParams, showCentosVersions];
+
+    try {
+      return await fetchFn(...fetchArgs);
+    } catch (error) {
+      return { results: [] };
     }
-  }, [hasAccess, apiParams, showCentosVersions]);
+  }, [
+    hasAccess,
+    JSON.stringify(apiParams),
+    showCentosVersions,
+    typeof fetchCustomOSes !== 'undefined',
```

Comment:
I think we discussed this part, but I am still questioning if this can be left as
```suggestion
    fetchCustomOSes,
```

Reply:
When it is left as it is, the tests were not happy.

Reply:
So we should update tests?

Reply:
I have made the necessary changes. It seems the tests are fixed with better handling of setData.